dotnet/src/Experimental/Process.LocalRuntime/LocalStep.cs

```diff
@@ -197,7 +197,7 @@ internal virtual async Task HandleMessageAsync(LocalMessage message)
         {
             this._logger?.LogError("Error in Step {StepName}: {ErrorMessage}", this.Name, ex.Message);
             eventName = $"{targetFunction}.OnError";
-            eventValue = ex.Message;
+            eventValue = ex;
```

Comment:
One thing to consider is that for distributed runtimes such as Dapr, we won't be able to use the exception object because they are generally not serializable. 

That said, it probably makes sense to allow the local runtime to provide as much info as it can... up to you if you want to standardize across the runtimes or not.

Reply:
That's a good point.  I did consider this.  I coudl've sworn the `Exception` class was designed to be able to serialize exceptions _across the wire_.

https://github.com/microsoft/referencesource/blob/master/mscorlib/system/exception.cs#L39

![image](https://github.com/user-attachments/assets/73d60231-71db-457e-baf7-cffb4e4f6720)

I've even seen code-analysis rules that enforce this for `Exception` subclasses.  I'm certainly tempted to let this ride until we discover where and why the pattern breaks down.  May make sense to revisit as a distributed runtime is being built.  What do you think?

I also wonder if in the distributed it may be an anti-pattern to handle the exception on a different service-node.  That is, it may be virtually unanimous that the exception is handled within the same runtime in which it was raised...at least the primary response.